src/tiled/objectsdock.cpp

```diff
@@ -401,6 +405,45 @@ void ObjectsView::selectedObjectsChanged()
     }
 }
 
+void ObjectsView::setColumnVisibility(bool visible)
+{
+    QAction *action = qobject_cast<QAction*>(sender());
+    if (action)
+        header()->setSectionHidden(action->data().toInt(), !visible);
+}
+
+void ObjectsView::showCustomMenu(const QPoint &point)
+{
+    Q_UNUSED(point)
+    QMenu contextMenu(this);
+    contextMenu.addActions(mActions);
+    contextMenu.exec(QCursor::pos());
+}
+
+void ObjectsView::updateColumnVisibilityActions()
+{
+    hideExtraColumns();
+    mActions.clear();
```

Comment:
This leaks all the QAction instances. However, as per another comment, I think you don't need `mActions` at all.